### ISSUE TITLE
Build failure: openvino

### ISSUE THREAD

**Issue by devusb** [NixOS/nixpkgs#282124]
### Steps To Reproduce

1. build *openvino*

### Build log

```
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/matmul/brgemm_matmul_reorders.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/matmul/brgemm_matmul_utils.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/matmul/jit_uni_sparse_matmul.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/prelu/jit_prelu_backward.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/prelu/jit_prelu_base_kernel.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/prelu/jit_prelu_forward.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/prelu/jit_prelu_reduction_kernel.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/prelu/jit_prelu_utils.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/prelu/jit_uni_prelu_backward_kernel.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/prelu/jit_uni_prelu_forward_kernel.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/rnn/brgemm_cell_common_bwd.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/rnn/brgemm_cell_common_fwd.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/rnn/brgemm_cell_common_reorders.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/rnn/brgemm_cell_common_utils.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/rnn/jit_brgemm_transpose_single_row.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/rnn/jit_diff_weights_peephole.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/rnn/jit_gates_reduction.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/rnn/rnn_brgemm_utils.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/shuffle/jit_uni_shuffle.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/shuffle/jit_uni_shuffle_kernel.cpp.o
[ 18%] Building CXX object src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeFiles/dnnl_cpu_x64.dir/utils/jit_io_helper.cpp.o
[ 18%] Built target dnnl_cpu_x64
make: *** [Makefile:156: all] Error 2
```

### Additional context

Also failing on [Hydra](https://hydra.nixos.org/build/246279025/nixlog/1).
Seems to fail at different parts of the build -- as early as 6% or as late as 18% (that I've seen).

### Notify maintainers

@tfmoraes 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.0, NixOS, 24.05 (Uakari), 24.05.20240115.c3e128f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.19.2`
 - nixpkgs: `/nix/store/11zbgb8j7wnnccbbjcq0q556h28g7p4r-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by eclairevoyant** [comment]
Actual error below:

```
In file included from /build/source/thirdparty/ade/sources/ade/include/ade/passes/topological_sort.hpp:17,
                 from /build/source/thirdparty/ade/sources/ade/source/topological_sort.cpp:7:
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp:101:10: error: 'uintptr_t' in namespace 'std' does not name a type
  101 |     std::uintptr_t m_srcGraph;
      |          ^~~~~~~~~
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp:22:1: note: 'std::uintptr_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   21 | #include "typed_metadata.hpp"
  +++ |+#include <cstdint>
   22 |
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp: In member function 'const ade::Graph& ade::ConstTypedGraph<Types>::getCGraph() const':
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp:106:53: error: 'm_srcGraph' was not declared in this scope
  106 |         return *reinterpret_cast<const ade::Graph*>(m_srcGraph);
      |                                                     ^~~~~~~~~~
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp: In constructor 'ade::ConstTypedGraph<Types>::ConstTypedGraph(const ade::Graph&)':
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp:122:9: error: class 'ade::ConstTypedGraph<Types>' does not have any field named 'm_srcGraph'
  122 |         m_srcGraph(reinterpret_cast<std::uintptr_t>(&graph))
      |         ^~~~~~~~~~
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp:122:42: error: 'uintptr_t' in namespace 'std' does not name a type
  122 |         m_srcGraph(reinterpret_cast<std::uintptr_t>(&graph))
      |                                          ^~~~~~~~~
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp:122:37: note: 'std::uintptr_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
  122 |         m_srcGraph(reinterpret_cast<std::uintptr_t>(&graph))
      |                                     ^~~
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp: In constructor 'ade::ConstTypedGraph<Types>::ConstTypedGraph(const ade::ConstTypedGraph<OtherTypes ...>&)':
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp:130:9: error: class 'ade::ConstTypedGraph<Types>' does not have any field named 'm_srcGraph'
  130 |         m_srcGraph(other.m_srcGraph)
      |         ^~~~~~~~~~
[  5%] Building CXX object thirdparty/protobuf/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/map.cc.o
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp: In instantiation of 'ade::Graph& ade::TypedGraph<Types>::getGraph() const [with Types = {ade::passes::TopologicalSortData}]':
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp:238:16:   required from 'ade::Graph::NodesListRange ade::TypedGraph<Types>::nodes() [with Types = {ade::passes::TopologicalSortData}; ade::Graph::NodesListRange = ade::util::Range::MapRange<ade::util::Range::IterRange<__gnu_cxx::__normal_iterator<std::shared_ptr<ade::Node>*, std::vector<std::shared_ptr<ade::Node> > > >, ade::Graph::HandleMapper>]'
/build/source/thirdparty/ade/sources/ade/source/topological_sort.cpp:39:40:   required from here
/build/source/thirdparty/ade/sources/ade/include/ade/typed_graph.hpp:167:53: error: 'const class ade::TypedGraph<ade::passes::TopologicalSortData>' has no member named 'm_srcGraph'
  167 |         return *reinterpret_cast<ade::Graph*>(this->m_srcGraph);
      |                                               ~~~~~~^~~~~~~~~~
make[2]: *** [thirdparty/ade/sources/ade/CMakeFiles/ade.dir/build.make:300: thirdparty/ade/sources/ade/CMakeFiles/ade.dir/source/topological_sort.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2208: thirdparty/ade/sources/ade/CMakeFiles/ade.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

probably a common gcc13 issue.

**Comment by eclairevoyant** [comment]
Even after fixing the gcc 13 issue, still getting errors like the following:

```
In member function 'operator()',
    inlined from '_FUN' at /build/source/src/common/preprocessing/src/ie_preprocess_gapi_kernels.cpp:176:16:
/build/source/src/common/preprocessing/src/ie_preprocess_gapi_kernels_simd_impl.hpp:731:23: error: inlining failed in call to 'always_inline' 'splitRowImpl': target specific option mismatch
  731 | CV_ALWAYS_INLINE void splitRowImpl(isa_tag_t, const T* in, std::array<T*, chs>& outs, const int length) {
      |                       ^
/build/source/src/common/preprocessing/src/ie_preprocess_gapi_kernels.cpp:178:44: note: called from here
  178 |             splitRowImpl<tag, uint8_t, chs>(t, in, outs, length);
      |                                            ^
In member function 'operator()',
    inlined from '_FUN' at /build/source/src/common/preprocessing/src/ie_preprocess_gapi_kernels.cpp:185:16:
/build/source/src/common/preprocessing/src/ie_preprocess_gapi_kernels_simd_impl.hpp:731:23: error: inlining failed in call to 'always_inline' 'splitRowImpl': target specific option mismatch
  731 | CV_ALWAYS_INLINE void splitRowImpl(isa_tag_t, const T* in, std::array<T*, chs>& outs, const int length) {
      |                       ^
/build/source/src/common/preprocessing/src/ie_preprocess_gapi_kernels.cpp:189:42: note: called from here
  189 |             splitRowImpl<tag, float, chs>(t, inT, outT, length);
      |                                          ^
In member function 'operator()',
    inlined from '_FUN' at /build/source/src/common/preprocessing/src/ie_preprocess_gapi_kernels.cpp:176:16:
/build/source/src/common/preprocessing/src/ie_preprocess_gapi_kernels_simd_impl.hpp:731:23: error: inlining failed in call to 'always_inline' 'splitRowImpl': target specific option mismatch
  731 | CV_ALWAYS_INLINE void splitRowImpl(isa_tag_t, const T* in, std::array<T*, chs>& outs, const int length) {
      |                       ^
/build/source/src/common/preprocessing/src/ie_preprocess_gapi_kernels.cpp:178:44: note: called from here
  178 |             splitRowImpl<tag, uint8_t, chs>(t, in, outs, length);
      |                                            ^
```

Seems like some issue with passed flags, but I don't think I'll have a chance to dig

**Comment by devusb** [comment]
Closed by #285536.

**Comment by owenstake** [comment]
You need to add the this
```c
#include <cstdint>
```
